### ISSUE TITLE
Improve DDP tutorial

### ISSUE THREAD

**Issue by rohan-varma** [pytorch/pytorch#65357]
## 📚 Documentation

Current [DDP tutorial](https://pytorch.org/tutorials/intermediate/ddp_tutorial.html) covers initializing distributed environment, DDP, checkpointing, and combining with model parallelism. To cover all of the recent features we've added to distributed training, integration with other parts of PyTorch training stack, and more accurately represent production training use cases we could enhance or create a new tutorial that:

1. Showcases how to initialize DDP with PyTorch elastic/torch.distributed.run
2. Demonstrates how to seamlessly scale from single node/laptop to cluster of machines
3. Cover a variety of useful features in DDP, such as support for unused parameter training, gradient compression, static graph, etc
4. Show how to do performance analysis/tuning
5. Show how to do inference in distributed setting and aggregate results as https://github.com/pytorch/pytorch/issues/64905 requests. 

Overall this will make the DDP tutorial more comprehensive and similar to real-world training use cases.

**Comment by vadimkantorov** [comment]
Also https://github.com/pytorch/pytorch/issues/64636
